azuki-script-generation/src/main/kotlin/com/anaplan/engineering/azuki/script/generation/ScriptGenerationCheck.kt

```diff
@@ -2,32 +2,44 @@ package com.anaplan.engineering.azuki.script.generation
 
 import com.anaplan.engineering.azuki.core.system.Check
 
-sealed interface ScriptGenerationCheck : Check
+interface ScriptGenerationCheck<E : ScriptGenerationEnvironment> : Check {
 
-/**
- * A basic, non-composable script generation check that does not need an environment.
- */
-interface BasicScriptGenerationCheck : ScriptGenerationCheck {
-
-    fun getCheckScript(): String
+    /**
+     * Gets the script for this check without trying to compose it.
+     * May fail if the check is impossible to express as DSL without composition.
```

Comment:
❓ How would it fail? Would it return nonsense or throw an exception? If the latter, should the interface specify which kind of exception? Or should it return a `Result`, like `resolveComposedCheck` below?

Reply:
Ah yeah, it would throw an exception, likely `IllegalStateException`.

In some of the code I've been writing to test this, I have a specific `MustCompose` object that fails with a stock exception if `getCheckScript` is called on it - the idea is you would _always_ specify it as `MustCompose.composeAs { ... }`.  I'm not sure if I could import it into Azuki since it'd become generic over `E`, but I'll see.

Reply:
I've upstreamed an `onlyComposable` function (and made `getCheckScript` carry `@Throws(IllegalStateException::class`) per the above.  I don't really want to make this return a `Result`, because it throwing an exception is somewhat exceptional (it indicates that a check that only makes sense to compose somehow slipped through and became uncomposable, which either means a catastrophically badly written/generated scenario _or_ a deficiency in the adapter)